### PR TITLE
fix markers if they're in db

### LINKED ISSUE #65: Handle dark bramble map and hud markers correctly
we need to use fog markers for dark bramble

### PR DESCRIPTION
**Pull request by bucketon**
I decided to just check the names of the goal points since there's only one case where the path is shared between two points. Currently I cannot figure out how to control the color of all the generated markers, but that's a bug I'm willing to live with, for now at least.

### CODE REVIEW

File: NomaiGrandPrix/NomaiGrandPrix.cs [hunk 1]
```diff
@@ -271,13 +271,46 @@ private void InitMapMarker()
...
+            {
+                outerWarp = GameObject.Find("DB_EscapePodDimension_Body/Sector_EscapePodDimension/Interactables_EscapePodDimension/OuterWarp_EscapePod")
+                    .GetComponent<OuterFogWarpVolume>();
+            }
+            else if (SpeedrunState.GoalPoint.Value.internalId.Equals("Spawn_Vessel"))
+            {
+                outerWarp = GameObject.Find("DB_VesselDimension_Body/Sector_VesselDimension/Interactables_VesselDimension/OuterWarp_Vessel")
+                    .GetComponent<OuterFogWarpVolume>();
+            }
+
```
Comment (bkirz) [inline NomaiGrandPrix/NomaiGrandPrix.cs]: wdyt about moving the goal point and game object path mappings into a Dictionary to reduce duplication? something like this:
```csharp
// Put this in a const outside the method def:
private static Dictionary<string, string> GameObjectPathsBySpawnPointID = new Dictionary<string, string>
{
    ["SpawnPoint_ElsinoreSeed"] = "DB_Elsinore_Body/Sector_ElsinoreDimension/Interactables_ElsinoreDimension/OuterWarp_Elsinore",
    ["SpawnPoint_EscapePod"] = "DB_EscapePodDimension_Body/Sector_EscapePodDimension/Interactables_EscapePodDimension/OuterWarp_EscapePod",
    ["SpawnPoint_PioneerCamp"] = "DB_PioneerDimension_Body/Sector_PioneerDimension/Interactables_PioneerDimension/OuterWarp_Pioneer",
    ["SpawnPoint_NomaiGrave"] = "DB_EscapePodDimension_Body/Sector_EscapePodDimension/Interactables_EscapePodDimension/OuterWarp_EscapePod",
    ["Spawn_Vessel"] = "DB_VesselDimension_Body/Sector_VesselDimension/Interactables_VesselDimension/OuterWarp_Vessel"
};

// Replace the if-else chain with:
OuterFogWarpVolume outerWarp = null;
var spawnPointId = SpeedrunState.GoalPoint.Value.internalId;

string gameObjectPath;
if (GameObjectPathsBySpawnPointID.TryGetValue(spawnPointId, out gameObjectPath))
{
    outerWarp = GameObject.find(gameObjectPath).GetComponent<OuterFogWarpVolume>();
}
```
This code sample might use language features we don't have access to based on the version of the .NET Framework we're using, but the general idea applies.

Comment (bkirz) [reply NomaiGrandPrix/NomaiGrandPrix.cs]: I forgot to say that this is non-blocking, sorry!

**Review by bkirz — APPROVED** [review]
One non-blocking refactoring suggestion, otherwise LGTM!

File: NomaiGrandPrix/NomaiGrandPrix.cs (hunk 1 above)
Comment (bucketon) [reply NomaiGrandPrix/NomaiGrandPrix.cs]: Yeah that makes sense to me. I was so excited to get it working I didn't do it that way to start but I can add that before merge.

Comment (bucketon) [reply NomaiGrandPrix/NomaiGrandPrix.cs]: fixed!

**Review by bkirz — APPROVED** [review]

**Pull request merged**